components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -628,10 +625,51 @@ func nbNameExists(client client.Client, nbName string, namespace string) bool {
 	return true
 }
 
+// predNBPodIsLabeled filters pods not containing the "notebook-name" label key
+func predNBPodIsLabeled() predicate.Funcs {
+	// Documented at
+	// https://github.com/kubernetes-sigs/controller-runtime/blob/ce8bdd3d81ab410ff23255e9ad3554f613c5183c/pkg/predicate/predicate_test.go#L884
+	checkNBLabel := func(label string) func(object client.Object) bool {
+		return func(object client.Object) bool {
+			_, labelExists := object.GetLabels()[label]
+			return labelExists
+		}
+	}
+
+	return predicate.NewPredicateFuncs(checkNBLabel("notebook-name"))
+}
+
+// predNBEvents filters events not coming from Pod or STS, and coming from
+// unknown NBs
+func predNBEvents(r *NotebookReconciler) predicate.Funcs {
+	checkEvent := func() func(object client.Object) bool {
+		return func(object client.Object) bool {
+			event := object.(*corev1.Event)
+			nbName, err := nbNameFromInvolvedObject(r.Client, &event.InvolvedObject)
+			if err != nil {
+				return false
+			}
+			return isStsOrPodEvent(event) && nbNameExists(r.Client, nbName, object.GetNamespace())
+		}
+	}
+
+	predicates := predicate.NewPredicateFuncs(checkEvent())
+
+	// Do not reconcile when an event gets deleted
+	predicates.DeleteFunc = func(e event.DeleteEvent) bool {
+		return false
+	}
+
+	return predicates
+}
+
+// SetupWithManager sets up the controller with the Manager.
 func (r *NotebookReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	builder := ctrl.NewControllerManagedBy(mgr).
 		For(&v1beta1.Notebook{}).
 		Owns(&appsv1.StatefulSet{}).
+		Owns(&corev1.Pod{}, builder.WithPredicates(predNBPodIsLabeled())).
```

Comment:
Something that is not clear to me yet: previously we had the `mapFn` , which was `handler.ToRequestsFunc`, that was responsible for converting the Pod to a namespaced event with the name/namespace of the owner notebook [and triggered `Reconcile`].

Now that we only have `Owns`, what namespaced event is going to trigger the `Reconcile` function when an update to a Pod is detected?

Reply:
I misunderstood the `mapFn` and `eventToRequest` functions logic so events were not being generated, apologies.

I have updated the PR to add these two functions back, adapting them to the the controller-runtime version -> https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/handler#example-EnqueueRequestsFromMapFunc